client/src/Admin/SitesForm.js

```diff
@@ -116,29 +123,25 @@ function SitesForm() {
               </label>
               <input type="text" className="form-control" id="website" name="website" onChange={onChange} value={data.website} />
             </div>
-            <div className="dropdown">
-              <button
-                className="btn btn-secondary dropdown-toggle"
-                type="button"
-                id="dropdownMenuButton"
-                data-toggle="dropdown"
-                aria-haspopup="true"
-                aria-expanded="false">
-                Nutrition Partner
-              </button>
-              <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
-                <Link style={{ textDecoration: 'none' }} to="/partners/new">
-                  <div className="dropdown-item">
-                    <i className="fa fa-plus phone" aria-hidden="true"></i>
-                    Add a new partner
-                  </div>
-                </Link>
-                {partners &&
-                  partners.map((partner, index) => (
-                    <div className="dropdown-item" onClick={() => updatePartner(index)}>
+            <div className="mb-3">
+              <label className="form-label">nutrition partners</label>
+              <div>
+                {partners?.map((partner) => (
+                  <div key={`partner-${id}`} className="form-check">
+                    <input
+                      type="checkbox"
+                      className="form-check-input"
+                      id={`partner-${partner.id}`}
+                      name="NutritionPartnerIds"
+                      value={partner.id}
+                      onChange={updateAssociation}
+                      checked={data.NutritionPartnerIds.includes(partner.id)}
```

Comment:
Every checkbox has a unique _id_, but the same _name_, which is the name of the array property in the data object storing associated ids. This allows the generic updateAssociation function to work properly onChange. See how we tell the box whether it is checked or not by looking in the array for the id.